packages/core/src/2d/assembler/TiledSpriteAssembler.ts

```diff
@@ -0,0 +1,288 @@
+      count = count % 1 >= stretch ? Math.ceil(count) : Math.floor(count);
+      if (count === 0) {
+        posColumn.push(expectHeight * bottom, fixedBottom, height - fixedTop, height - expectHeight * (1 - top));
+        uvColumn.push(spriteUVs[0].y, spriteUVs[1].y, NaN, NaN, spriteUVs[2].y, spriteUVs[3].y);
+      } else if (count <= Basic2DBatcher.MAX_VERTEX_COUNT / posRow.length / 4 - 3) {
+        heightScale = height / (fixedTop + fixedBottom + count * fixedCenterH);
+        posColumn.push(expectHeight * bottom * heightScale, fixedBottom * heightScale);
+        uvColumn.push(spriteUVs[0].y, spriteUVs[1].y, spriteUVs[1].y);
+        for (let i = 0, l = count - 1; i < l; i++) {
+          posColumn.push(fixedBottom + (i + 1) * fixedCenterH * heightScale);
+          uvColumn.push(spriteUVs[2].y, spriteUVs[1].y);
+        }
+        posColumn.push(height - fixedTop * heightScale, height - expectHeight * (1 - top) * heightScale);
+        uvColumn.push(spriteUVs[2].y, spriteUVs[2].y, spriteUVs[3].y);
+      } else {
+        posRow.length = uvRow.length = 0;
+        posRow.push(width * left, width * right);
+        uvRow.push(spriteUVs[0].x, spriteUVs[3].x);
+        posColumn.push(height * bottom, height * top);
+        uvColumn.push(spriteUVs[0].y, spriteUVs[3].y);
+      }
+    }
+  }
+
+  private static _calculateContinuous(
+    sprite: Sprite,
+    width: number,
+    height: number,
+    posRow: number[],
+    posColumn: number[],
+    uvRow: number[],
+    uvColumn: number[]
+  ) {
+    const { zeroTolerance } = MathUtil;
+    const { border } = sprite;
+    const spriteUVs = sprite._getUVs();
+    const spritePositions = sprite._getPositions();
+    const { x: left, y: bottom } = spritePositions[0];
+    const { x: right, y: top } = spritePositions[3];
+    const { width: expectWidth, height: expectHeight } = sprite;
+    const fixedLeft = expectWidth * border.x;
+    const fixedBottom = expectHeight * border.y;
+    const fixedRight = expectWidth * border.z;
+    const fixedTop = expectHeight * border.w;
+    if (fixedLeft + fixedRight >= width) {
+      const widthScale = width / (fixedLeft + fixedRight);
+      posRow.push(
+        expectWidth * left * widthScale,
+        fixedLeft * widthScale,
+        width - expectWidth * (1 - right) * widthScale
+      );
+      uvRow.push(spriteUVs[0].x, spriteUVs[1].x, spriteUVs[2].x, spriteUVs[3].x);
+    } else {
+      const fixedCenterW = expectWidth - fixedLeft - fixedRight;
+      const count = fixedCenterW > zeroTolerance ? (width - fixedLeft - fixedRight) / fixedCenterW : 0;
+      if (count === 0) {
+        posRow.push(expectWidth * left, fixedLeft, width - fixedRight, width - expectWidth * (1 - right));
+        uvRow.push(spriteUVs[0].x, spriteUVs[1].x, NaN, NaN, spriteUVs[2].x, spriteUVs[3].x);
+      } else if (count <= Basic2DBatcher.MAX_VERTEX_COUNT / 4 / 4 - 4) {
+        posRow.push(expectWidth * left, fixedLeft);
```

Comment:
All SpriteRenderer object with tiled draw mode will use Basic2DBatcher.MAX_VERTEX_COUNT / 4 / 4，should be static, and Basic2DBatcher.MAX_VERTEX_COUNT / 16 is better

Reply:
Done.